### PR TITLE
ACIR to raw R1CS

### PR DESCRIPTION
**Pull request by ilitteri**

### CODE REVIEW

File: src/gnark_backend_glue/acir_to_r1cs.rs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+use crate::acvm;
+use anyhow::Result;
+
+// AcirCircuit and AcirArithGate are R1CS-friendly structs.
+//
+// The difference between these structures and the ACIR structure that the compiler uses is the following:
+// - The compilers ACIR struct is currently fixed to bn254
+// - These structures only support arithmetic gates, while the compiler has other
+// gate types. These can be added later once the backend knows how to deal with things like XOR
+// or once ACIR is taught how to do convert these black box functions to Arithmetic gates.
```
Comment (Loco1982) [inline src/gnark_backend_glue/acir_to_r1cs.rs:11]:
```suggestion
// or once ACIR supports converting these black box functions to Arithmetic gates.
```

File: src/gnark_backend_glue/acir_to_r1cs.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+pub struct RawGate {
+    pub mul_terms: Vec<(Fr, acvm::Witness, acvm::Witness)>,
+    pub add_terms: Vec<(Fr, acvm::Witness)>,
+    pub constant_term: Fr,
+}
+
+impl RawR1CS {
+    pub fn new(acir: acvm::Circuit, values: Vec<acvm::FieldElement>) -> Result<Self> {
+        // Currently non-arithmetic gates are not supported
+        // so we extract all of the arithmetic gates only
```
Comment (Loco1982) [inline src/gnark_backend_glue/acir_to_r1cs.rs:30]:
```suggestion
        // so we extract only all of the arithmetic gates
```

File: src/gnark_backend_glue/mod.rs [hunk 3]
```diff
@@ -1,5 +1,32 @@
 use acvm::{acir::circuit::Circuit, FieldElement};
 
+mod acir_to_r1cs;
+mod serialize;
+
+// Arkworks's types are generic for `Field` but Noir's types are concrete and
+// its value depends on the feature flag.
```
Comment (Loco1982) [inline src/gnark_backend_glue/mod.rs:7]:
```suggestion
// their value depends on the feature flag.
```

**Review by Loco1982 — COMMENTED** [review]
Minor syntax changes

**Comment by ilitteri** [comment]
[Map ACIR to R1CS](https://trello.com/c/WZNouXZI/13-map-acir-to-r1cs)

**Review by jrchatruc — APPROVED** [review]

**Pull request merged**